packages/nimble-components/src/wafer-map/specs/features/rendering.md

```diff
@@ -0,0 +1,189 @@
+
+Example:
+
+```TS
+    const functionName = 'createWaferMap';
+    performance.mark(`${functionName}_start`);
+    const wafermap = document.createElement('nimble-wafer-map');
+    document.body.appendChild(wafermap);
+    performance.mark(`${functionName}_end`);
+    performance.measure(functionName, `${functionName}_start`, `${functionName}_end`);
+```
+
+The example shows how custom markers can be set in the tested code. The result of these user marks can be observed inside lighthouse like in the screenshot bellow.
+
+![User Timing and measures](./resources/user-timing-marks-and-measures.PNG)
+
+We wish to extend the existing lighthouse CI testing to include the wafermap component and record various metrics about the performance it has when rendering multiple generated data loads. They will include data ingestion, rendering time, rerendering when user events get triggered and data changes.
+
+These records will be made in a separate web page from the existing performance tests to avoid polluting the results. They will not be preserved over time, each CI record being available for download after build.
+
+Our target goals are:
+
+1. For wafers >200k and <1m die
+
+    1. Initial rendering: <1 second
+    1. Zoom/pan actions: <500ms
+    1. Configuration changes that don't modify the dataset (e.g. show labels):
+       <500ms
+    1. Configuration changes that modify the dataset (e.g. replace layers): <1 second
+
+1. For wafers >1m and <10m die
+    1. Initial rendering: <10 seconds
+    1. Zoom/pan actions: <1 second
+    1. Interactions that don't modify the dataset (e.g. show labels): <1 second
+    1. Interactions that modify the dataset (e.g. replace layers): <10 seconds
+
+## Alternative Implementations / Designs
+
+### Alternative Data Structures and Interfaces
+
+The above mentioned structure can also be implemented as an [apache arrow](https://arrow.apache.org/docs/js/index.html) table with columns and metadata.
+
+Another option is to break each object property as a separate attribute for the wafer map component.
+
+### Alternative Rendering
+
+Alternatives to the described rendering are splitting the data and canvas and using multiple threads to enhance performance even more.
+
+### Alternative Size Limitations
+
+In order to provide a smooth user experience, if the existing event interaction rendering is not sufficiently fast we would implement a more instantaneous response to zoom gestures using bitmap scaling of the canvas while the renderer is updating the final display.
+
+Another alternative to remove the size limits is to create a point reduction algorithm which will create a reduced set of dies fit to the size limits creating an average, median or other approximation. This approach introduces additional complexity to the rendering and is a lossy compression algorithm, so it's implementation should be carefully considered.
+
+## Open Issues
```

Comment:
I think I remember someone suggesting that we should figure out the Blazor and Angular APIs for the new data representation early. Do you know how those frameworks would represent typed arrays? Would there be any performance concerns with transforming data from the framework's representation to the typed array?